ui/page/dcrdex/dex_onboarding_page.go

```diff
@@ -880,76 +918,189 @@ func (pg *DEXOnboarding) HandleUserInteractions() {
-				if utils.EditorsNotEmpty(pg.serverURLEditor.Editor) {
-					serverURL = pg.serverURLEditor.Editor.Text()
-					serverCert = []byte(pg.serverCertEditor.Editor.Text())
-				} else {
-					pg.serverURLEditor.SetError(values.String(values.StrDEXServerAddrWarning))
+				pg.exchange = xc
+
+				pg.bondAssets = make(map[libutils.AssetType]*core.BondAsset)
+				capitalize := cases.Upper(language.Und)
+				var supportedBondAssets []libutils.AssetType
+				for _, asset := range xc.BondAssets {
+					assetSym := dex.BipIDSymbol(asset.ID)
+					assetType := libutils.AssetType(capitalize.String(assetSym))
+					if !pg.WL.AssetsManager.IsAssetTypeSupported(assetType) {
+						continue
+					}
+					supportedBondAssets = append(supportedBondAssets, assetType)
+					pg.bondAssets[assetType] = asset
+				}
+
+				pg.noSupportedBondAsset = len(supportedBondAssets) == 0
+				if pg.noSupportedBondAsset {
+					pg.notifyError(values.StringF(values.StrNoSupportedBondAsset, pg.selectedServerURL))
 					return
 				}
+
+				pg.currentStep = onboardingPostBond
+				// TODO: pg.bondSourceWalletSelector should be an asset type
+				// selector so users can easily create missing wallets and fund
+				// it with the required bond amount.
+				pg.bondSourceWalletSelector = components.NewWalletAndAccountSelector(pg.Load, supportedBondAssets...).
+					Title(values.String(values.StrSelectWallet)).
+					WalletSelected(func(wm *load.WalletMapping) {
+						if err := pg.bondSourceAccountSelector.SelectFirstValidAccount(wm); err != nil {
+							log.Error(err)
+						}
+					})
+				pg.bondSourceAccountSelector = components.NewWalletAndAccountSelector(pg.Load, supportedBondAssets...).
+					Title(values.String(values.StrSelectAcc)).
+					AccountSelected(func(a *sharedW.Account) {
+						pg.bondAccountHasEnough()
+					}).AccountValidator(func(a *sharedW.Account) bool {
+					return !a.IsWatchOnly
+				})
+				pg.bondSourceAccountSelector.HideLogo = true
+				if err := pg.bondSourceAccountSelector.SelectFirstValidAccount(pg.bondSourceWalletSelector.SelectedWallet()); err != nil {
+					log.Error(err)
+				}
+
+				pg.bondStrengthEditor.Editor.SetText(fmt.Sprintf("%d", minimumBondStrength))
+				pg.newTier = minimumBondStrength
 			}
 
-			// TODO: Validate server is reachable and connect.
-			_ = serverURL
-			_ = serverCert
+			if pg.isLoading {
+				break
+			}
 
-			pg.currentStep = onboardingPostBond
-			pg.bondSourceWalletSelector = components.NewWalletAndAccountSelector(pg.Load /*, supportedAssets...  TODO: Use assets provided by selected DEX server. */).
-				Title(values.String(values.StrSelectWallet)).
-				WalletSelected(func(wm *load.WalletMapping) {
-					if err := pg.bondSourceAccountSelector.SelectFirstValidAccount(wm); err != nil {
-						log.Error(err)
+			dexInitialized := pg.dexc.IsInitialized()
+			if !dexInitialized {
+				go func() {
+					pg.isLoading = true
+					pg.dexPass = []byte(pg.passwordEditor.Editor.Text())
```

Comment:
Why do we need to cache the pass? Do we zero the bytes when we no longer need it?

Reply:
> Why do we need to cache the pass?

When users provide their pass via the pass modal, we shouldn't request the pass again while the user is on the onboarding page so we need to cache it so we can access it after the pass modal is closed.